src/tiled/thumbnailrenderer.cpp

```diff
@@ -98,7 +98,7 @@ static QRectF cellRect(const MapRenderer &renderer,
 static QRect computeMapRect(const MapRenderer &renderer)
 {
     // Start with the basic map size
-    QRectF rect(QPointF(0, 0), renderer.mapSize());
+    QRectF rect(QPointF(0, 0), renderer.mapBoundingRect().size());
```

Comment:
the thumbnail renderer renders the tiles for the tile stamp dock. So you can simply add some tiles there to test it.